functionaltests/steps_test.go

```diff
@@ -35,65 +35,63 @@ import (
 )
 
 type testStepsRunner struct {
-	namespace    string
-	startVersion ecclient.StackVersion
-	steps        []testStep
+	DataStreamNamespace string
+	Steps               []testStep
 }
 
 func (r testStepsRunner) Run(t *testing.T) {
-	if r.namespace == "" {
-		r.namespace = "default"
+	if r.DataStreamNamespace == "" {
+		r.DataStreamNamespace = "default"
 	}
 
 	start := time.Now()
 	ctx := context.Background()
-	tf := initTerraformRunner(t)
 
-	t.Log("------ cluster setup ------")
-	esCfg := createCluster(t, ctx, tf, *target, r.startVersion.String())
-	esc := createESClient(t, esCfg)
-	kbc := createKibanaClient(t, ctx, esc, esCfg)
-	g := createAPMGenerator(t, ctx, esc, esCfg)
-	t.Logf("time elapsed: %s", time.Since(start))
-
-	env := testStepEnv{
-		namespace: r.namespace,
-		version:   r.startVersion,
-		tf:        tf,
-		gen:       g,
-		kbc:       kbc,
-		esc:       esc,
-	}
-
-	docCounts, err := getDocsCountPerDS(t, ctx, esc)
-	require.NoError(t, err)
-
-	currentRes := testStepResult{DocCounts: docCounts}
-	for _, step := range r.steps {
+	env := testStepEnv{dsNamespace: r.DataStreamNamespace}
+	currentRes := testStepResult{}
+	for _, step := range r.Steps {
 		currentRes = step.Step(t, ctx, &env, currentRes)
 		t.Logf("time elapsed: %s", time.Since(start))
 	}
 }
 
 type testStepResult struct {
-	DocCounts esclient.DataStreamsDocCount
+	DSDocCount esclient.DataStreamsDocCount
 }
 
 type testStepEnv struct {
-	namespace string
-	version   ecclient.StackVersion
-	tf        *terraform.Runner
-	gen       *gen.Generator
-	kbc       *kbclient.Client
-	esc       *esclient.Client
+	dsNamespace string
+	version     ecclient.StackVersion
+	tf          *terraform.Runner
+	gen         *gen.Generator
+	kbc         *kbclient.Client
+	esc         *esclient.Client
 }
 
 type testStep interface {
 	Step(t *testing.T, ctx context.Context, e *testStepEnv, previousRes testStepResult) testStepResult
 }
 
+type createStep struct {
+	DeployVersion ecclient.StackVersion
+}
+
+func (c createStep) Step(t *testing.T, ctx context.Context, e *testStepEnv, _ testStepResult) testStepResult {
+	t.Log("------ cluster setup ------")
+	e.tf = initTerraformRunner(t)
+	esCfg := createCluster(t, ctx, e.tf, *target, c.DeployVersion.String())
+	e.esc = createESClient(t, esCfg)
+	e.kbc = createKibanaClient(t, ctx, e.esc, esCfg)
+	e.gen = createAPMGenerator(t, ctx, e.esc, esCfg)
+
+	docCount := getDocsCountPerDS(t, ctx, e.esc)
+	return testStepResult{DSDocCount: docCount}
+}
+
+var _ testStep = ingestionStep{}
+
 type ingestionStep struct {
```

Comment:
nit: this step should probably be named `ingestStep` for coherence with other steps.